src/transformation/visitors/literal.ts

```diff
@@ -137,13 +138,10 @@ const transformObjectLiteralExpression: FunctionVisitor<ts.ObjectLiteralExpressi
 };
 
 const transformArrayLiteralExpression: FunctionVisitor<ts.ArrayLiteralExpression> = (expression, context) => {
-    const values = expression.elements.map(element =>
-        lua.createTableFieldExpression(
-            ts.isOmittedExpression(element) ? lua.createNilLiteral(element) : context.transformExpression(element),
-            undefined,
-            element
-        )
+    const filteredElements = expression.elements.map(e =>
+        ts.isOmittedExpression(e) ? ts.createIdentifier("undefined") : e
```

Comment:
Have you checked that `ts.createIdentifier("undefined")`  is actually transformed to `nil`?

Reply:
That is tested over at array.spec.ts

Reply:
Why not just use actual `nil` here

Reply:
> Why not just use actual nil here

It's TS AST, not Lua

> That is tested over at array.spec.ts

Have you actually checked it's output? Since `undefined` is not defined this test might pass even if it wouldn't be correctly transformed

Reply:
Added c8ebe4e to use spread and check for the expected undefined value

Reply:
> Why not just use actual `nil` here

I intended `flattenSpreadExpressions` to transform TS expressions without caring where those expressions are

`OmittedExpression` has no transformer so it will break if one gets to `flattenSpreadExpressions`